crates/cxx-qt-gen/test_outputs/inheritance.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+
+namespace rust::cxxqtlib1 {
+template<typename T>
+class CxxQtThread;
+}
+
+class MyObject;
+using MyObjectCxxQtThread = ::rust::cxxqtlib1::CxxQtThread<MyObject>;
+
+#include "cxx-qt-gen/inheritance.cxx.h"
+
+class MyObject : public QAbstractItemModel
+{
+  Q_OBJECT
+
+public:
+  explicit MyObject(QObject* parent = nullptr);
+  ~MyObject();
+  MyObjectRust const& unsafeRust() const;
+  MyObjectRust& unsafeRustMut();
+  ::std::unique_ptr<MyObjectCxxQtThread> qtThread() const;
+
+public:
+  Q_INVOKABLE QVariant data(QModelIndex const& _index,
+                            ::std::int32_t _role) const override;
+  Q_INVOKABLE bool hasChildren(QModelIndex const& _parent) const override;
+  template<class... Args>
```

Comment:
can you open a separate bug about considering if we want to use `friend` as discussed before in the future. Or if just making them public like this is also "fine". And in that bug put a link to #329 so we don't forget the syntax for that :sweat_smile:

Reply:
#452